robot-core/src/main/java/org/obolibrary/robot/IOHelper.java

```diff
@@ -1198,6 +1211,64 @@ public void saveContext(File file) throws IOException {
     writer.close();
   }
 
+  /**
+   * Determine if a string is a CURIE. Note that a valid CURIE is not always a valid QName. Adapted
+   * from:
+   *
+   * @see org.semanticweb.owlapi.io.XMLUtils#isQName(CharSequence)
+   * @param s Character sequence to check
+   * @return true if valid CURIE
+   */
+  public static boolean isValidCURIE(CharSequence s) {
+    if (s == null || 0 >= s.length()) {
+      // string is null or empty
+      return false;
+    }
+    boolean inLocal = false;
+    for (int i = 0; i < s.length(); ) {
+      int codePoint = Character.codePointAt(s, i);
+      if (codePoint == ':') {
+        if (inLocal) {
+          // Second colon - illegal
+          return false;
+        }
+        inLocal = true;
+      } else {
+        if (!inLocal) {
+          // Check for valid NS characters
+          if (!XMLUtils.isXMLNameStartCharacter(codePoint)) {
+            return false;
+          }
+        } else {
+          // Check for valid local characters
+          if (!XMLUtils.isXMLNameChar(codePoint)) {
+            return false;
+          }
+        }
+      }
+      i += Character.charCount(codePoint);
+    }
+    return true;
+  }
+
+  /**
+   * Determine if the short form of an IRI contains invalid characters.
+   *
+   * @param iri IRI to check
+   * @return true if no invalid characters found
+   */
+  public static boolean shortFormIsValid(IRI iri) {
```

Comment:
Is valid what? CURIE? QName? String?

Reply:
It validates that the short form does not contain invalid characters. I guess `shortFormContainsInvalidCharacters` (returns true on invalid) or something like that sounded too long to me but I can change it.

Reply:
My point is that there are no invalid characters in general, without saying what the standard is. There are invalid characters for a CURIE, or a QName, or an IRI, or a Python function name.

I'll think about it a little more and make changes myself.